### ISSUE TITLE
JS Error: too much recursion

### ISSUE THREAD

**Issue by bfsgr** [aunetx/blur-my-shell#557]
I don't know exactly how to reproduce this issue, but sometimes the extension causes an error that gnome-shell reports as "JS ERROR: too much recursion". It may or may not be causing an issue with workspace switching [reported upstream](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7519).

OS: Fedora 39 on Wayland

```txt
JS ERROR: too much recursion              createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:730:46
                                          get@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/conveniences/settings.js:51:59
                                          _log@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:304:13
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:78:28
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52
                                          enable/wac_proto._prepareWorkspaceSwitch@file:///home/bfsgr/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/components/overview.js:79:52

```

**Comment by FOUR-BITS-28** [comment]
I'm also experiencing this issue. I noticed it first when I unlocked the computer and then the workspace switch animation began to not work. Consequently, the windows from other workspaces started overlapping with each other every time I switch. Upon checking the logs from `journalctl -b`, I saw the same "JS ERROR: too much recursion" that gnome-shell reported. Logging out and logging in will temporarily fix the issue, but will come back after unlocking the computer.
